cmake/CPM.cmake

```diff
@@ -464,6 +464,80 @@ function(cpm_check_git_working_dir_is_clean repoPath gitTag isClean)
 
 endfunction()
 
+# method to add PATCH_COMMAND to CPM_ARGS_UNPARSED_ARGUMENTS for PATCHES files.
+function(cpm_add_patches)
+  # Return if no patch files are supplied.
+  if(NOT ARGN)
+    return()
+  endif()
+
+  # Provide a small warning.
+  if(CMAKE_VERSION VERSION_LESS "3.20")
+    message(
+      AUTHOR_WARNING
+        "Versions of CMake less than 3.20 may need to supply patch files as absolute paths."
+    )
+  endif()
+
+  # Find the patch program.
+  find_program(PATCH_EXECUTABLE patch)
+  if(WIN32 AND NOT PATCH_EXECUTABLE)
+    # The Windows git executable is distributed with patch.exe. Find the path to the executable, if
+    # it exists, then search `../../usr/bin` for patch.exe.
+    find_package(Git QUIET)
+    if(GIT_EXECUTABLE)
+      if(CMAKE_VERSION VERSION_GREATER_EQUAL "3.20")
+        cmake_path(GET GIT_EXECUTABLE PARENT_PATH extra_search_path)
+        cmake_path(GET extra_search_path PARENT_PATH extra_search_path)
+      else()
+        get_filename_component(extra_search_path ${GIT_EXECUTABLE} DIRECTORY)
+        get_filename_component(extra_search_path ${extra_search_path} DIRECTORY)
+        get_filename_component(extra_search_path ${extra_search_path} DIRECTORY)
+      endif()
```

Comment:
I am suspicious `get_filename_component` will soon be deprecated and eventually removed. I think we can simplify the code to use the older functions for now, but may still eventually need to update it to this. It may take years for it's removal though, so if your preference is for simpler code I'm fine with that.

Reply:
Upon further reflection, I believe CMake itself will eventually support `PATCHES` for `FetchContent()` and this is likely to happen BEFORE `get_filename_component()` is deprecated. I'll make the change and run some tests.

This is a functionality that I truly want to see in a release!

Reply:
Addressed in 523c14a1c8b2f9625c0d5c55d2d3deb3c529da74